pkg/koordlet/resourceexecutor/resctrl.go

```diff
@@ -0,0 +1,147 @@
+/*
+Copyright 2022 The Koordinator Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package resourceexecutor
+
+import (
+	"errors"
+	"fmt"
+	"os"
+	"path/filepath"
+	"strconv"
+	"strings"
+
+	"k8s.io/klog/v2"
+
+	"github.com/koordinator-sh/koordinator/pkg/koordlet/util/system"
+)
+
+const ErrResctrlDir = "resctrl path or file not exist"
+const CacheIdIndex = 2
+
+func NewResctrlReader() ResctrlReader {
```

Comment:
What if the resctrl is enabled during the koordlet's running? Perhaps we can check the support status inside the reading methods.

Reply:
In my design, if both the CPU and OS support resctrl, data will not be actually [collected](https://github.com/koordinator-sh/koordinator/blob/e98f7eb7fc2b533580bf6399ff8b4642e7b7143f/pkg/koordlet/metricsadvisor/collectors/resctrl/resctrl_collector.go#L97-L114) after enabling the resctrl collector plugin without mounting; data will be reported after `mount`.

Reply:
OK